panther_driver/src/panther_can.py

```diff
@@ -18,327 +18,337 @@ def __setattr__(self, name, value):
-        self.runtime_stat_flag = [0, 0]
-        self.temperature = 0.0
-        self.fault_flags = 0
-        self.script_flags = 0
+        self._can_node = canopen.RemoteNode(can_node_id, eds_file)
 
-        self.can_node = canopen.RemoteNode(can_node_id, eds_file)
+    @property
+    def connection_error(self) -> bool:
+        if self._can_node.nmt.state != 'OPERATIONAL':
+            return True
 
-    def __setattr__(self, name: str, value: Any) -> None:
-        super().__setattr__(name, value)
-        if name not in ['_last_time_callback', 'can_node']:
-            self._last_time_callback[name] = rospy.Time.now()
+        return rospy.Time.now() - self._heartbeat_timestamp > rospy.Duration(
+            self.HEARTBEAT_TIMEOUT_S
+        )
 
     @property
-    def last_time_callback(self) -> Dict[str, rospy.Time]:
-        return self._last_time_callback
+    def driver_temperature(self) -> float:
+        with self._lock:
+            try:
+                self._driver_temperature = float(self._can_node.sdo['Qry_TEMP'][1].raw)
+            except canopen.SdoCommunicationError:
+                pass
 
+        return self._driver_temperature
 
-class PantherCAN:
-    def __init__(self, eds_file: str, can_interface: str) -> None:
-        self._lock = Lock()
-        self._network = canopen.Network()
+    @property
+    def battery_data(self) -> List[float]:
+        with self._lock:
+            try:
+                # division by 10 is needed according to documentation
+                amps = sum(
+                    [
+                        float(self._can_node.sdo['Qry_BATAMPS'][wheel].raw) / 10.0
+                        for wheel in self._wheels
+                    ]
+                )
+                voltage = (
+                    float(self._can_node.sdo['Qry_VOLTS'][ControllerChannels.VOLT_CHANNEL].raw)
+                    / 10.0
+                )
 
-        self._motor_controllers = [
-            MotorController(1, eds_file),  # front
-            MotorController(2, eds_file),  # rear
-        ]
+                self._battery_data = [amps, voltage]
+            except canopen.SdoCommunicationError:
+                pass
 
-        self._network.connect(channel=can_interface, bustype='socketcan')
+        return self._battery_data
+
+    def setup(self, network: canopen.Network) -> None:
+        network.add_node(self._can_node)
+        self._can_node.nmt.add_hearbeat_callback(self._heartbeat_cb)
+
+    def write_wheels_enc_velocity(self, vel: List[float]) -> None:
+        with self._lock:
+            for i, wheel in enumerate(self._wheels):
+                try:
+                    self._can_node.sdo['Cmd_CANGO'][wheel].raw = vel[i]
+                except:
+                    pass
 
-        self._network.add_node(self._motor_controllers[0].can_node)
-        self._network.add_node(self._motor_controllers[1].can_node)
+    def _heartbeat_cb(self, nmt_state: int) -> None:
```

Comment:
```suggestion
    def _heartbeat_cb(self, _) -> None:
```
We can indicate that this value wont be used inside callback